ft_memmove.c

```diff
@@ -0,0 +1,60 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_memmove.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: kkubik <kkubik@student.42prague.com>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/01/12 21:31:42 by kkubik            #+#    #+#             */
+/*   Updated: 2023/01/12 21:59:13 by kkubik           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+void	*ft_memmove(void *dest, void *src, size_t n);
+
+void	*ft_memmove(void *dest, void *src, size_t n)
+{
+	size_t	i;
+	char	*srcp;
+	char	*destp;
+	char	*temp;
+
+	i = 0;
+	srcp = (char *)src;
+	destp = (char *)dest;
+	//I do not know how to do this without malloc
+	temp = (char *)malloc(sizeof(char) * n);
```

Comment:
Maybe smth like this? Please recheck me.

> Material from here - https://www.geeksforgeeks.org/memmove-in-cc/
> memmove() copies the data first to an intermediate buffer, then from the buffer to destination.

```
void  *ft_memmove(void *dest, void *src, size_t n)
{
  size_t  i;
  char  *srcp;
  char  *destp;
  char  temp;

  i = 0;
  srcp = (char *)src;
  destp = (char *)dest;
  //I do not know how to do this without malloc
  //temp = (char *)malloc(sizeof(char) * n);
  if (!dest && !src)
    return (NULL);
  while (i < n)
  {
    temp = srcp[i];
     destp[i] = temp;
    i++;
  }

  return (dest);
}
```

Reply:
Brilliant idea to use temp as one symbol <3